packages/design-system-react-native/src/components/TextButton/TextButton.tsx

```diff
@@ -0,0 +1,243 @@
+/* eslint-disable @typescript-eslint/prefer-nullish-coalescing */
+import { useTailwind } from '@metamask/design-system-twrnc-preset';
+import React, { useMemo, useState } from 'react';
+import type { GestureResponderEvent } from 'react-native';
+import { View } from 'react-native';
+
+import type { SpinnerProps } from '../../temp-components/Spinner';
+import Spinner from '../../temp-components/Spinner';
+import type { IconProps, IconColor } from '../Icon';
+import Icon from '../Icon';
+import Text from '../Text/Text';
+import type { TextProps } from '../Text/Text.types';
+import { DEFAULT_TEXTBUTTON_PROPS } from './TextButton.constants';
+import type { TextButtonProps } from './TextButton.types';
+import {
+  generateTextButtonContainerClassNames,
+  generateTextButtonTextClassNames,
+} from './TextButton.utilities';
+
+const TextButton = ({
+  children,
+  textProps,
+  isLoading = DEFAULT_TEXTBUTTON_PROPS.isLoading,
+  loadingText,
+  spinnerProps,
+  startIconName,
+  startIconProps,
+  startAccessory,
+  endIconName,
+  endIconProps,
+  endAccessory,
+  isDisabled = DEFAULT_TEXTBUTTON_PROPS.isDisabled,
+  isInverse = DEFAULT_TEXTBUTTON_PROPS.isInverse,
+  onPress,
+  onPressIn,
+  onPressOut,
+  onLongPress,
+  twClassName,
+  accessibilityRole,
+  accessibilityLabel,
+  style,
+  ...props
+}: TextButtonProps) => {
+  const [isPressed, setIsPressed] = useState(false);
+  const tw = useTailwind();
+
+  // Styling calculation for state changes
+  const twContainerClassNames = useMemo(() => {
+    return generateTextButtonContainerClassNames({
+      isPressed,
+      isLoading,
+      isDisabled,
+      twClassName,
+    });
+  }, [isPressed, isLoading, isDisabled, twClassName]);
+
+  const twTextClassNames = useMemo(() => {
+    return generateTextButtonTextClassNames({
+      isPressed,
+      isInverse,
+      isLoading,
+    });
+  }, [isPressed, isInverse, isLoading, textProps]);
```

Comment:
Should `textProps` be removed from the dependency array here? It doesn't seem to be used in the `twTextClassNames` useMemo hook. Not sure why linting didn't pick this up 🤔

Reply:
good catch! it was used at some point for the calculation but is no longer needed so I'll remove it